### PR TITLE
spark job view refine (part 1)

### PR DESCRIPTION
**Pull request by voidfunction**
local cache for job view
clean job view http server

### CODE REVIEW

File: Utils/hdinsight-node-common/resources/htmlResources/hdinsight/job/html/sparkjobview.js [hunk 1]
```diff
@@ -67,19 +91,19 @@ $(function () {
...
-    $("#sparkEventButton").click(function () {
-        JobUtils.openSparkEventLog(projectId, typeof appId == 'undefined' ? "" : appId.toString());
+    $('#sparkEventButton').click(function () {
+        JobUtils.openSparkEventLog(projectId, typeof appId === 'undefined' ? "" : appId.toString());
     });
 
-    $("#livyLogButton").click(function() {
-        JobUtils.openLivyLog(typeof appId == 'undefined' ? "" : appId.toString());
+    $('#livyLogButton').click(function() {
+        JobUtils.openLivyLog(typeof appId === 'undefined' ? "" : appId.toString());
```
Comment (djlan) [inline Utils/hdinsight-node-common/resources/htmlResources/hdinsight/job/html/sparkjobview.js]: >"" [](start = 60, length = 2)

NIT, all use the same, single quote or double quote?

File: Utils/hdinsight-node-common/resources/htmlResources/hdinsight/job/html/sparkjobview.js [hunk 2]
```diff
@@ -243,122 +271,82 @@ function setAMcontainer() {
...
 function setDiagnosticsLog() {
-    if (appId.substr(0, 5) == "local") {
+    if (appId.substr(0, 5) === "local") {
         $("#errorMessage").text("No Yarn Error Message");
     } else {
-        getMessageAsync(localhost + projectId + "/cluster/apps/" + appId + "?restType=yarn", function (s) {
-            var object = JSON.parse(s);
-            var message = object.app.diagnostics;
-            if (message == 'undefined' || message == "") {
+        getMessageAsync("/apps?appId=" + spark.appId + 'yarn', function (s) {
```
Comment (djlan) [inline Utils/hdinsight-node-common/resources/htmlResources/hdinsight/job/html/sparkjobview.js]: >getMessageAsync("/apps?appId=" + spark.appId + 'yarn', function (s) { [](start = 8, length = 69)

getMessageAsync now need type as second parameter?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java [hunk 3]
```diff
@@ -21,26 +21,46 @@
...
+            return sparkApplicationMap.get(clusterName)
+                    .stream()
+                    .filter(app-> app.getId().equalsIgnoreCase(appId))
+                    .findFirst()
+                    .orElse(null);
+        }
+        return null;
+    }
 
     public synchronized static void registerJovViewNode(@NotNull String uuid, @NotNull IClusterDetail clusterDetail) {
```
Comment (djlan) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java]: >registerJovViewNode(@NotNull String uuid [](start = 36, length = 40)

They key of jobViewPanelMap is uuid or clustername?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java [hunk 4]
```diff
@@ -21,26 +21,46 @@
...
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public class JobViewManager {
-    private static Map<String, IClusterDetail> jobViewPanelMap = new HashMap<String, IClusterDetail>();
+    private static Map<String, IClusterDetail> jobViewPanelMap = new HashMap<>();
+    private static Map<String, List<Application>> sparkApplicationMap = new HashMap<>();
+
+    public synchronized static void registerApplications(@NotNull String clusterName, @NotNull List<Application> apps) {
```
Comment (djlan) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java]: >synchronized [](start = 11, length = 12)

How about using HashTable and remove all synchronized?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java (hunk 3 above)
Comment (djlan) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java]: Please rename the parameter name. Seems like it should be clustername

File: Utils/hdinsight-node-common/resources/htmlResources/hdinsight/job/html/sparkjobview.js (hunk 1 above)
Comment (voidfunction) [inline Utils/hdinsight-node-common/resources/htmlResources/hdinsight/job/html/sparkjobview.js]: single quote as consistency

File: Utils/hdinsight-node-common/resources/htmlResources/hdinsight/job/html/sparkjobview.js (hunk 2 above)
Comment (voidfunction) [inline Utils/hdinsight-node-common/resources/htmlResources/hdinsight/job/html/sparkjobview.js]: yes.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java (hunk 3 above)
Comment (voidfunction) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java]: It's cluster name now. and cluster name is a unique key in one Azure environment.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java (hunk 4 above)
Comment (voidfunction) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java]: Sounds good. updated.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java [hunk 5]
```diff
@@ -21,29 +21,50 @@
...
+            return sparkApplicationMap.get(clusterName)
+                    .stream()
+                    .filter(app-> app.getId().equalsIgnoreCase(appId))
+                    .findFirst()
+                    .orElse(null);
+        }
+        return null;
+    }
+
+    public static void registerJovViewNode(@NotNull String uuid, @NotNull IClusterDetail clusterDetail) {
```
Comment (djlan) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/JobViewManager.java]: >String uuid [](start = 52, length = 11)

Change all uuid to clusterName?

**Comment by djlan** [comment]
:shipit:

**Pull request merged**